### DISCUSSION TITLE
How to show the horizontal scroll bar to aggrid?

### DISCUSSION THREAD

**Discussion by snoman88** [zauberzeug/nicegui#3292] [Q&A] [answered]
### Question

When I add a long data to a cell, aggrid displays "abide...". But I want that column auto-resized and horizontal scroll bar under the aggrid. How can I do that?

```py
rowData = [{"name": str(i), "path": str(i)*200} for i in range(100)]
ui.aggrid(
{"columnDefs": [
    {"field": "name", "headerName": "File"},
    {"field": "path", "headerName": "Path"}],
"rowSelection": "multiple",
"rowData": rowData,
"autoSizeStrategy": "fitCellContents",
},
html_columns=[0])
ui.run()
```

result is:
<img width="902" alt="스크린샷 2024-06-28 오후 1 54 42" src="https://github.com/zauberzeug/nicegui/assets/101552104/5d2c6724-95ef-4ecd-80ae-87870a8d0903">

**Comment by python-and-novella** [comment]


**Reply by snoman88** [reply]
Thank you for your answer.
But I want to show the horizontal scroll bar not wrap.

**Reply by python-and-novella** [reply, marked as answer]
```python3
from nicegui import ui,app

rowData = [{"name": str(i), "path": str(i)*200} for i in range(100)]
grid = ui.aggrid(
{"columnDefs": [
    {"field": "name", "headerName": "File"},
    {"field": "path", "headerName": "Path"}],
"rowSelection": "multiple",
"rowData": rowData,
"autoSizeStrategy": "fitCellContents",
},
html_columns=[0])
#ui.timer(3,lambda :grid.run_grid_method('autoSizeAllColumns'),once=True)
ui.button('autoSizeAllColumns',on_click=lambda :grid.run_grid_method('autoSizeAllColumns'))
ui.button('sizeColumnsToFit',on_click=lambda :grid.run_grid_method('sizeColumnsToFit'))
ui.run()
```
![image](https://github.com/zauberzeug/nicegui/assets/16256910/d69346a5-51df-4856-85be-3e005877b1a0)

Is it as expected?

**Reply by snoman88** [reply]
Your image is exactly what I want, but I couldn't get the same result when I ran your code. I pressed buttons but nothing happened.

**Reply by python-and-novella** [reply]
It wokrs as expected after upgrading aggrid to `31.0.0` .
I'm not sure if the upgraded version will cause other issues.

**Reply by snoman88** [reply]
How can I upgrade aggrid? pip install -U nicegui was not working.

**Reply by python-and-novella** [reply]
Path is `.venv\Lib\site-packages\nicegui\elements\lib\aggrid\ag-grid-community.js` or similarly in your Python's dir.
Replace it manually with aggrid official release dist package.
You'd better wait nicegui 2.0 when it was released and try to upgrade it by `pip install -U`,.It is the right way.
![image](https://github.com/zauberzeug/nicegui/assets/16256910/9725e09b-5620-4961-a424-8114b9988e89)

**Reply by snoman88** [reply]
Thank you for your kind and patient responses. Your answers were very helpful for me.

**Reply by jakechevrier** [reply]
how do you achieve this when using a pandas dataframe?
